src/Language/ASKEE/Exposure/Interpreter.hs

```diff
@@ -147,8 +147,8 @@ data EvalRead = EvalRead
   , erGetFileFn   :: EvalReadFileFn
   }
 
-type EvalReadFileFn  = FilePath -> IO LBS.ByteString
-type EvalWriteFileFn = FilePath -> LBS.ByteString -> IO ()
+type EvalReadFileFn  = FilePath -> IO (Either Text LBS.ByteString)
```

Comment:
The type `Either Text ByteString` smells a bit funny, as both are textual representations without any connotations of "error-ness". I'd suggest using `Either SomeException LBS.ByteString` instead to make this more explicit, leaving the choice of how to render the `SomeException` (to `Text` or otherwise`) to consumers.

Reply:
Oh thanks that's a good idea

Reply:
Ah, the trick is that we don't get a `SomeException` in the Donu case, as we'll just get an error message from the client.

Reply:
Bah. In that case, I relent.